arduino/cores/board.go

```diff
@@ -139,3 +139,43 @@ func (b *Board) GeneratePropertiesForConfiguration(config string) (*properties.M
 	}
 	return b.GetBuildProperties(fqbn.Configs)
 }
+
+// IsBoardMatchingIDProperties returns true if the board match the given
+// identification properties
+func (b *Board) IsBoardMatchingIDProperties(query *properties.Map) bool {
```

Comment:
I'd write a unit test for this function.

Reply:
Done, and CI already found a bug! LOL